### PR TITLE
fix(workload): expose the Kubernetes config flags on workload wait

### LINKED ISSUE #6144: fix(workload): expose Kubernetes config flags on workload wait
> 🤖 Generated by the Daily AI Engineer

## Reproduction and evidence

`pkg/client/kubectl/commands.go` creates `genericclioptions.NewConfigFlags(true)` in `CreateWaitCommand` and passes it to upstream `wait.NewCmdWait`, but never registers those flags on the Cobra command. This differs from the other kubectl-backed commands, whose shared `customizeCommand` calls `configFlags.AddFlags(cmd.Flags())`.

On current `main` (`deeaf6ddabd3dc895ee12f72d6871e969f53f6bf`):

```text
$ go run . workload wait -n ksail-intercept-demo --for=condition=Ready pod/example --timeout=2m
unknown shorthand flag: 'n' in -n
```

The generated `workload wait` reference exposes `--for` and `--timeout`, but not `--namespace`, `--context`, or `--kubeconfig`. Existing tests only assert the wait-specific flags.

## Audience and impact

People using KSail to wait for a resource outside the current context's default namespace cannot select that namespace or another kubeconfig context. They must reconfigure their current context or leave KSail for a separate kubectl command, which makes otherwise self-contained KSail workflows fail.

## Expected behavior

`ksail workload wait` should expose and honor the same Kubernetes config flags as the other kubectl-backed workload commands, including namespace, context, and kubeconfig selection.

## Acceptance criteria

- [ ] `workload wait --help` exposes `--namespace`/`-n`, `--context`, and `--kubeconfig`.
- [ ] The selected namespace, context, and kubeconfig reach upstream wait execution.
- [ ] Tests cover flag registration and argument parsing so the current omission cannot regress.
- [ ] The generated command reference and snapshots are regenerated.

## Rough size

Small bug fix with command tests and generated-doc updates.

### PR DESCRIPTION
**Pull request by devantler**
> 🤖 Generated by the Agentic Engineer

## Why

`ksail workload wait` could only wait on resources in the current context's default namespace: passing `-n`, `--context` or `--kubeconfig` failed with "unknown shorthand flag", unlike every other kubectl-backed workload command. Anyone waiting on a resource elsewhere had to switch context by hand or drop out of KSail for a separate kubectl call.

## What

`workload wait` now accepts and honours the same Kubernetes config flags as its sibling commands, so a namespace, context or kubeconfig can be selected inline. The generated reference page, chat docs and help snapshots reflect the new flags; nothing changes for callers that never passed them.

Fixes #6144

### CODE REVIEW

**Comment by devantler** [comment]
> 🤖 Generated by the Agentic Engineer

CI is green at `721d794f583974c9a3d9c26c5b4458b26791cad8` (60 checks; the one earlier red was a network transient in an unrelated unit test, re-run once and captured as #6901), so requesting the first current-head review while this stays a draft.

@coderabbitai review

**Comment by devantler** [comment]
> 🤖 Generated by the Agentic Engineer

The 21:17Z request was refused by CodeRabbit's per-hour chat limit (retry window stated as ~10 minutes); retrying once at the same green head `721d794f583974c9a3d9c26c5b4458b26791cad8`. A second refusal records `cr:no-gate` and the lane advances to Codex.

@coderabbitai review

**Comment by devantler** [comment]
> 🤖 Generated by the Agentic Engineer

CodeRabbit refused twice at this head (21:17Z per-hour chat limit, 21:28Z review rate limit with a 29-minute window), so `cr:no-gate@721d794f583974c9a3d9c26c5b4458b26791cad8` and the lane advances. Requesting the next provider at the same green head while this stays a draft.

@codex review

File: pkg/client/kubectl/commands.go [hunk 1]
```diff
@@ -295,6 +295,12 @@ func (c *Client) CreateWaitCommand(kubeConfigPath string) *cobra.Command {
 
 	waitCmd := wait.NewCmdWait(configFlags, c.ioStreams)
 
+	// Register the Kubernetes config flags (--namespace, --context, --kubeconfig, …)
+	// on the command, as every other kubectl-backed workload command does. NewCmdWait
+	// reads them through configFlags but does not add them itself, which left wait
+	// unable to target a namespace or context other than the current one (#6144).
+	configFlags.AddFlags(waitCmd.Flags())
```
[the remark this answers is not in the document]
Comment (devantler) [reply pkg/client/kubectl/commands.go:302]: > 🤖 Generated by the Agentic Engineer

Valid, and wider than this PR — fixed at the root rather than for `wait` alone.

Verified against the code: every kubectl-backed workload command is built through `newKubectlCommand`, which applies `wrapWithKubeconfigResolution`, and that hook sets the flag **only when the user did not change it**. So an explicit `--kubeconfig` reached client-go unresolved on `describe`, `explain`, `forward`, `get` and `logs` already on `main`; this PR extends the exposure to `wait` rather than creating it. The rule you cite is in the repository's own conventions: every user-supplied path argument must go through `fsutil.EvalCanonicalPath`.

The fix canonicalizes the explicit value inside that shared hook, so all six commands are covered by one change. A path that does not exist yet still resolves (`EvalCanonicalPath` falls back to the parent directory), so an unreadable kubeconfig keeps failing downstream with client-go's own message instead of a new error here; the unset path is unchanged, including the resolved default shown in `--help`.

Four tests pin it: a symlinked `--kubeconfig` resolves to its target, a relative one becomes absolute, an unset one still takes the config-derived path and default, and a missing file still resolves. The first two fail on the previous behaviour and pass with the fix.

Fixed in `e0f4d733d1a9d47ff690082a0c8060d6d7b24d88`.

**Comment by devantler** [comment]
> 🤖 Generated by the Agentic Engineer

CI is green at `5e97ff976a78c1eab5454b5dd40c318078d4181a` (76 checks, no failures). Since Codex's review at `721d794f` this head adds a merge of `main` with the generated artifacts re-run, the root-cause fix for Codex's P1 (an explicitly passed `--kubeconfig` is now canonicalized in the shared hook, covering all six kubectl-backed workload commands) with four tests, and a lint-directive placement fix. Requesting a review at this head while the PR stays a draft.

@coderabbitai review

File: pkg/cli/cmd/workload/kubeconfig_flag_test.go [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+	t.Parallel()
+
+	dir := t.TempDir()
+	missing := filepath.Join(dir, "absent-kubeconfig")
+
+	cmd, flag := newKubeconfigFlagCommand(t, missing, true)
+
+	require.NoError(t, workload.ExportResolveKubeconfigFlag(cmd, flag, "/config/derived/path"))
+
+	assert.Contains(t, flag.Value.String(), "absent-kubeconfig")
```
[the remark this answers is not in the document]
Comment (devantler) [reply pkg/cli/cmd/workload/kubeconfig_flag_test.go]: > 🤖 Generated by the Agentic Engineer

Valid — fixed in `cae5a964`. The assertion now compares the whole canonical path (`filepath.Join(canonicalDir, "absent-kubeconfig")` against the symlink-resolved temp dir), so a resolution that picked a different parent with the same basename no longer passes.

File: pkg/cli/cmd/workload/shared.go [hunk 3]
```diff
@@ -201,6 +200,42 @@ func wrapWithKubeconfigResolution(cmd *cobra.Command) {
...
+// downstream with client-go's own message rather than here.
+func resolveKubeconfigFlag(
+	child *cobra.Command,
+	kubeconfigFlag *pflag.Flag,
+	resolvedPath string,
+) error {
+	value := resolvedPath
+
+	if child.Flags().Changed("kubeconfig") {
+		canonical, err := fsutil.EvalCanonicalPath(kubeconfigFlag.Value.String())
```
[the remark this answers is not in the document]
Comment (devantler) [reply pkg/cli/cmd/workload/shared.go]: > 🤖 Generated by the Agentic Engineer

Valid, and a regression this PR introduced — fixed in `cae5a964`. `filepath.Abs("")` returns the working directory, so `--kubeconfig ""` would have been handed to client-go as a directory to load instead of falling back to its own default resolution; before this PR an explicitly empty value was left untouched.

`resolveKubeconfigFlag` now returns early when the supplied value is empty, restoring that behaviour, and `TestResolveKubeconfigFlag_ExplicitEmptyStaysEmpty` pins it — it fails without the guard and passes with it.

**Comment by devantler** [comment]
> 🤖 Generated by the Agentic Engineer

CI is green at `cae5a964` — 76 checks, no failures, including the full Docker system-test matrix. Both of your findings are fixed at this head: the missing-path test asserts the whole canonical path, and an explicitly empty `--kubeconfig` is left empty rather than becoming the working directory, pinned by a test that fails without the guard. Replies are on their threads.

Requesting a review at this head while the PR stays a draft.

@coderabbitai review

**Comment by devantler** [comment]
> 🤖 Generated by the Agentic Engineer

## Readiness at `cae5a964`

**Programmatically tested.** CI is green at this head: 76 checks, no failures, including the full Docker system-test matrix and `CI - Required Checks`. The canonicalization is pinned by six tests — symlinked, relative, unset, missing-file, explicitly-empty, and the resolved default shown in `--help` — of which three fail on the behaviour that preceded them.

**Reviewed.** CodeRabbit completed a review at this head (`Review completed`, summary refreshed at 05:47:55Z naming `cae5a964`) with no findings. Its two earlier findings and Codex's P1 are each fixed and answered; all three threads are resolved.

**Tried and evaluated as a user.** Built at this head and exercised against a real cluster:

- `ksail workload wait -n ksail-probe-ns …` → `Error from server (NotFound): namespaces "ksail-probe-ns" not found`. Before this PR the same command failed with `unknown shorthand flag: 'n'`, so the flag is now accepted and the namespace reaches the API server.
- `--help` lists `-n, --namespace`, `--context` and `--kubeconfig`, the last showing the resolved default path.
- `--kubeconfig <symlink>` and a relative `--kubeconfig ./link` both reach the API server, so canonicalization resolves them rather than handing client-go an unresolved path.
- `--kubeconfig ""` also reaches the API server, confirming the empty value still falls back to default resolution instead of being turned into the working directory.

Promoting.
